libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarResponseErrorMap.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import java.util.HashMap
+import okhttp3.ResponseBody
+import retrofit2.Response
+
+internal class TarResponseErrorMap(private val errorCodes: HashMap<Int, String>) {
+
+    companion object {
+        private const val TILES_ACCESS_TOKEN_ERROR_CODE = 402
+        private const val BOUNDING_BOX_ERROR_CODE = 422
+        private const val TILES_ACCESS_TOKEN_ERROR_MESSAGE =
+            "Unable to fetch tiles: Before you can fetch " + "routing tiles you must obtain an enterprise access token. Please contact us at support@mapbox.com"
+        private const val BOUNDING_BOX_ERROR_MESSAGE =
+            "Unable to fetch tiles: The bounding box you have " + "specified is too large. Please select a smaller box and try again."
+        private const val ERROR_MESSAGE_FORMAT = "Error code %s: %s"
+    }
+
+    init {
+        errorCodes[TILES_ACCESS_TOKEN_ERROR_CODE] = TILES_ACCESS_TOKEN_ERROR_MESSAGE
+        errorCodes[BOUNDING_BOX_ERROR_CODE] = BOUNDING_BOX_ERROR_MESSAGE
+    }
+
+    fun buildErrorMessageWith(response: Response<ResponseBody>): String {
+        var errorMessage = errorCodes[response.code()]
+        if (errorMessage == null) {
+            errorMessage = "Error code ${response.code()}: ${response.message()}"
```

Comment:
Could we fix `ERROR_MESSAGE_FORMAT` removing ` %s: %s` and use it here instead?

Reply:
done